addons/account/models/partner.py

```diff
@@ -424,8 +424,18 @@ def _get_company_currency(self):
     invoice_warn_msg = fields.Text('Message for Invoice')
     # Computed fields to order the partners as suppliers/customers according to the
     # amount of their generated incoming/outgoing account moves
-    supplier_rank = fields.Integer(default=0)
-    customer_rank = fields.Integer(default=0)
+    supplier_rank = fields.Integer(default=0, copy=False)
+    customer_rank = fields.Integer(default=0, copy=False)
+
+    def copy(self, default=None):
+        self.ensure_one()
+        # Set context according to ranks fields to simulate where was created original partner
```

Comment:
I don't get why we need to do that. Could you explain?

Reply:
But if you do that with the context, you'll be in trouble when duplicating a partner that is *both* customer and supplier: he'll only get the customer rank (because the context key will be set by both conditions; hence overridden by the second one). This is annoying, as the copied partner will then appear only in 'customers' filter. So, essentially, it's not really fixing the behavior of the copy.

To do that, you should rather add default values of 1 for both ranks in the default variable that is passed to the super call to copy ; so that both ranks are set to 1 in the case I described.